utils/timeline.py

```diff
@@ -86,14 +90,84 @@ def pull(self,
 
             # insert tweets into file
             tweets: List[dict] = response.data
+            has_refs: bool = 'referenced_tweets' in self.query_params.tweet_fields
+            inc_tweets: List[dict] = response.includes['tweets'] if 'tweets' in response.includes.keys() else None
+            inc_users: List[dict] = response.includes['users'] if 'users' in response.includes.keys() else None
+            # inc_media: List[dict] = response.includes['media'] if 'media' in response.includes.keys() else None
             if tweets:
+                
+                # Expansions parsing
+                if inc_tweets:
+                    ref_tweets = [twalc.Tweet(**tw).to_dict() for tw in inc_tweets]
+                if inc_users:
+                    authors = [twalc.User(**us).to_dict() for us in inc_users]
+                # if inc_media:
+                #     media = [twalc.Media(**md) for md in inc_media]
+                if has_refs:
+                    links = Timeline.__parse_tweet_links(tweets)
+
+                # Original Tweets Parsing
                 tweets = [twalc.Tweet(**tw).to_dict() for tw in tweets]
+                
+
+                # Expansions dataframes
+                if inc_tweets:
+                    df_refs   = pd.concat([df_refs, pd.DataFrame(ref_tweets)], axis = 0) if df_refs is not None else pd.DataFrame(ref_tweets)
```

Comment:
is there any harm in using `dtype=str` inside pd.DataFrame to force all fields to be treated as strings? That would sidestep the need for the fix floats function (and I think the dependency on python version >=3.8). I guess the public metrics are a little inefficient as strings but that doesn't seem very important.

Reply:
Attempted to pass dtype=str to the constructor, but in the pandas code, it seems like the values are coerced to floats before being forced to strings with this parameter, so the decimal points and scientific notation still remain with this approach.

Reply:
fix_floats removed